### PR TITLE
Chore: break social connections into snippets

### PR DESCRIPTION
**Pull request by faroceann**
## References
DXEC-163

## Script used
`articles/connections/parse.script.js`

```javascript
const fs = require("fs");
const path = require("path");
const _ = require("lodash");

const getInclude = (include, name) => {
  let text = fs.readFileSync(path.join(__dirname, include), {
    encoding: "UTF8",
  });
  text = text.replace(/\$\{idp\}/g, _.startCase(name))
  return text;
};

const splitter = (name, string) => {
  let prev;
  let curSection = [];
  let sections = [];
  let heading = [];
  const closing = "---";
  let reachedClose = false;
  let reachedOpen = false;
  const newDoc = [];
  string.split("\n").forEach((line, i) => {
    if (line === closing && !reachedOpen) {
      newDoc.push(line + "\n");
      reachedOpen = true;
      return;
    }
    if (line === closing && reachedOpen) {
      newDoc.push(line + "\n");
      reachedClose = true;
      return;
    }
    if (!reachedClose) {
      newDoc.push(line + "\n");
      return;
    }
    if (line.includes("<%= include('../_quickstart-links.md') %>")) {
      newDoc.push(line + "\n");
      return;
    }
    if (line.includes("<%= include(")) {
      console.warn("Include encountered: ", line);
      let includePath = line.split("'")[1].replace("../", "");
      if (!includePath.endsWith(".md")) {
        includePath += ".md";
      }
      const include = getInclude(includePath, name);
      curSection.push(include);
      return;
    }
    if (line.includes("## ") || line.includes("# ") || line.includes("### ")) {
      if (curSection.join("")) {
        sections.push(curSection.join(""));
      }
      if (curSection.length) {
        newDoc.push(`<%= include('../../../snippets/social/${name}/${sections.length - 1}') %> \n`);
      }
      curSection = [];
    } else {
      heading = [];
    }
    if (line) {
      curSection.push(line + '\n');
    }
    prev = line;
  });
  return { newDoc, sections };
};

const connections = fs.readdirSync(path.join(__dirname, "./social"));

connections.forEach((manifestPath) => {
  if (manifestPath.includes("github")) {
    return;
  }
  console.log(manifestPath);
  const manifest = fs.readFileSync(
    path.join(__dirname, "social", manifestPath),
    { encoding: "UTF8" }
  );
  const name = manifestPath.replace(".md", "")
  const split = splitter(name, manifest);
  const snippetDir = path.join(__dirname, "..", "..", "snippets", "social", name);
  if (!fs.existsSync(snippetDir)) {
    fs.mkdirSync(snippetDir);
  }
  split.sections.forEach((section, i) => {
    fs.writeFileSync(
      path.join(snippetDir, i + ".md"),
      section
    );
  });
  fs.writeFileSync(path.join(__dirname, "social", name + ".md"), split.newDoc.join(''));
  console.log(split);
});
```

## Outcome – snippets API social section returns 41 items
image

image

### CODE REVIEW

File: articles/connections/social/devkeys.md [hunk 1]
```diff
@@ -9,47 +9,6 @@ useCase:
     - customize-connections
     - add-idp
 ---
-
```
Comment (joshcanhelp) [inline articles/connections/social/devkeys.md]: I don't think this one needs to be converted 🤔

File: articles/connections/social/instagram.md [hunk 2]
```diff
@@ -15,50 +15,9 @@ useCase:
     - customize-connections
     - add-idp
 ---
-# Connect Apps to Instagram
```
Comment (joshcanhelp) [inline articles/connections/social/instagram.md]: This one is deprecated so we won't have a guide for it.

File: articles/connections/social/oauth2.md [hunk 3]
```diff
@@ -14,199 +14,11 @@ useCase:
     - customize-connections
     - add-idp
 ---
-# Connect Apps to Generic OAuth2 Authorization Servers
```
Comment (joshcanhelp) [inline articles/connections/social/oauth2.md]: This is for a custom connection and is for the extension so probably another one to omit

File: articles/connections/social/reprompt-permissions.md [hunk 4]
```diff
@@ -8,24 +8,5 @@ useCase:
     - customize-connections
     - add-idp
 ---
-
```
Comment (joshcanhelp) [inline articles/connections/social/reprompt-permissions.md]: Can probably leave this one out also

File: articles/connections/social/soundcloud.md [hunk 5]
```diff
@@ -15,44 +15,9 @@ useCase:
     - customize-connections
     - add-idp
 ---
-
```
Comment (joshcanhelp) [inline articles/connections/social/soundcloud.md]: This one is deprecated so we won't have a guide for it.

**Review by joshcanhelp — CHANGES_REQUESTED** [review]
Nice work on this, love that you wrote a script ... I probably would have copy/pasted like a caveman 😆 

In general, I would suggest leaving out anything that's not going to be used from the API. Also flagged a few of the deprecated/removed ones which, at thing point, probably should be deleted (cc @rkhoriander).

File: articles/connections/social/devkeys.md (hunk 1 above)
Comment (faroceann) [reply articles/connections/social/devkeys.md]: @joshcanhelp do you think it should be removed, or reverted?

Comment (faroceann) [reply articles/connections/social/devkeys.md]: Going to opt for reverting all of the deprecated/ unneeded conversions

**Review by joshcanhelp — DISMISSED** [review]
🙌  🙌  🙌

**Review by erjohnson — APPROVED** [review]

**Pull request merged**